### DISCUSSION TITLE
Duplicate icons

### DISCUSSION THREAD

**Discussion by setanarut** [elrumo/macOS_Big_Sur_icons_replacements#1086] [General]
I guess duplicate icons are not checked. An issue template can be made to report duplicate icons.
<img width="718" alt="Ekran Resmi 2021-02-08 05 27 44" src="https://user-images.githubusercontent.com/36481442/107169254-78744f80-69ce-11eb-8ae2-9cf3337d77d6.png">

**Comment by elrumo** [comment]
Good idea! I'll try to implement something to fix that

**Comment by rodgerwerner** [comment]
It seems this is working now as there don't seem to be any duplicates when searching for Transmission icons.
Question: I uploaded both an app icon and matching doc icon but I can only see the doc icon. Did my app icon get flagged somehow? Because I made it from scratch and it is different than all the others.

**Reply by setanarut** [reply]
document icons cannot be replaced from `Get Info`. I don't understand what the purpose of this is. It is necessary to rebuild the Application for document icons.

**Reply by rodgerwerner** [reply]
Although document icons cannot be replaced from Get Info, they can be replaced by other means.

**Reply by setanarut** [reply]
@rodgerwerner your document icon is broken. image sizes are wrong in ICNS.

https://user-images.githubusercontent.com/36481442/108126816-5740ec80-70bb-11eb-8d44-652f7312b88a.mov

**Reply by rodgerwerner** [reply]
First: I downloaded the icon from here and when viewed in Preview, all sizes are the correct image, the one I made.
Second: I assume you are seeing this because your icon cache isn't completely cleared. Did you restart Finder and the Dock processes? Or, did you restart macOS?

**Reply by setanarut** [reply]
> did you restart macOS?
Yes

You are right. After a while macOS changes the icns file. (Overwriting). I guess that's because I modified the App. Also the icon is different when I switch to macOS Light mode. 
1. i replaced /Applications/Transmission.app/Contents/Resources/TransmissionDocument.icns
2. Restart macOS

I found these;

[Clear Mac OS X's icon cache. Gist](https://gist.github.com/fabiofl/5873100)
[Q: How to clear Mac OSX icon cache](https://discussions.apple.com/thread/8441124)

**Reply by rodgerwerner** [reply]
Simply restarting macOS doesn't always rebuild the icon cache. For me, on a new M1 Mini, all I've had to do is kill the Dock process (it restarts on it's own) and Relaunch Finder with ⌥+right click on the Finder icon on the Dock. These will force a rebuilding of the icon cache.

I have tried deleting (renaming) the icon services etc with Terminal, but generally speaking that doesn't always work for me.

One other thing, you might want to try 'touching' the app with Terminal:

```
sudo su -
touch /Applications/Transmission.app
touch /Applications/Transmission.app/Contents/Info.plist
killall Dock
```

This should work on Transmission, but not on all apps. Basically, this will reload Transmission's file icons into the icon cache ... after you have replaced them.

**Reply by setanarut** [reply]
Relaunching Finder and `killall Dock` doesn't work for me.

I never use `sudo` as a principle.

**Reply by rodgerwerner** [reply]
I'm fresh out of ideas then, but it does sound like an icon caching issue from my perspective. Someone else might have ideas.

Regarding `sudo` , I understand.

**Reply by setanarut** [reply]
https://eclecticlight.co/2019/01/30/why-doesnt-that-documents-icon-display-properly-beyond-quicklook/

<img width="940" alt="getdocicon01" src="https://user-images.githubusercontent.com/36481442/108136888-ebb34b00-70cb-11eb-86e1-a1f4f93983de.png">
